### PR TITLE
[WasmFS] Fix standalone wasm printing, newline instead of 0

### PR DESCRIPTION
**Pull request by kripken**
0 works in some places but at least in wasmtime it does not.

### CODE REVIEW

File: system/lib/standalone/standalone.c [hunk 1]
```diff
@@ -176,11 +176,13 @@ imported__wasi_fd_write(__wasi_fd_t fd,
...
 
 static void wasi_write(__wasi_fd_t fd, char* buffer) {
-  struct __wasi_ciovec_t iov;
-  iov.buf = (uint8_t*)buffer;
-  iov.buf_len = strlen(buffer) + 1;
+  struct __wasi_ciovec_t iovs[2];
+  iovs[0].buf = (uint8_t*)buffer;
+  iovs[0].buf_len = strlen(buffer);
+  iovs[1].buf = (uint8_t*)"\n";
+  iovs[1].buf_len = 1;
```
Comment (sbc100) [inline system/lib/standalone/standalone.c:183]: Can you rename this method to `wasi_writeln` perhaps?  Or maybe `wasi_puts`?  took me a while to figure out this wasn't being called for every user-level write call.

**Review by sbc100 — APPROVED** [review]

**Review by tlively — APPROVED** [review]
Would it make sense to not append any new characters and just pass the user data through untouched?

If not, LGTM 👍

File: system/lib/standalone/standalone.c (hunk 1 above)
Comment (kripken) [reply system/lib/standalone/standalone.c:184]: Good point, thanks. Renaming.

**Comment by kripken** [comment]
The old wasmtime fails here. I was testing on a newer one locally which is why I didn't notice before now.

Bump the version from 0.8.0 to 0.18.0. The newer version no longer has the bug, and the test should pass.

**Comment by kripken** [comment]
The new wasi now warns when sent an invalid clock_id. The extra warning makes the test fail. Bumping to latest 0.33.0, that now aborts the entire VM on an invalid clock_id. @sbc100 you follow WASI, is that correct? I can't seem to find online docs for what the WASI clock APIs should do (I find what the functions are, but not how they should handle errors). I would expect an invalid input to lead to an error code (0.8.0) not a printed warning (0.18.0) or a VM fault (0.33.0)...

**Comment by sbc100** [comment]
> The new wasmtime now warns when sent an invalid clock_id. The extra warning makes the test fail. Bumping to latest 0.33.0, that now aborts the entire VM on an invalid clock_id. @sbc100 you follow WASI, is that correct? I can't seem to find online docs for what the WASI clock APIs should do (I find what the functions are, but not how they should handle errors). I would expect an invalid input to lead to an error code (0.8.0) not a printed warning (0.18.0) or a VM fault (0.33.0)...

Hmm.. I don't remember now.   Do you want me to take a look at doing the wasmtime update as a separate PR?  Presumably we can fix the issue by not passing an invalid value for clock_id?

**Comment by kripken** [comment]
We could work around it, but that would increase code size which sounds unfortunate if it is a wasmtime bug...

I'll file an issue there.

**Comment by kripken** [comment]
Filed https://github.com/bytecodealliance/wasmtime/issues/3714

I guess let's hold off on this PR and/or upgrading wasmtime for now.

**Comment by sbc100** [comment]
I created a standalone PR work around the issue: https://github.com/emscripten-core/emscripten/pull/16091

It only effects STANDALONE_WASM mode so I don't think two extra conditional should worry us.

**Pull request merged**